python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -906,12 +911,66 @@ async def run_calibration_sequence(
 
             step = dict(
                 wavelength=exposure.wavelength,
+                dacValue=exposure.dacValue,
                 mtcamera_exposure_info=mtcamera_exposure_info,
             )
 
             calibration_summary["steps"].append(step)
         return calibration_summary
 
+    def generate_random_exposure_times(
+        self,
+        samples_per_bin: list[int],
+        bin_edges: list[list[float]],
+        dac_values: list[float],
+    ) -> tuple[list[float], list[float]]:
+        """
+        Generate a list of random exposure times based on bin edges and the
+        number of samples per bin.
+        Parameters
+        ----------
+        samples_per_bin : int
+            The number of random samples to generate within each bin.
+        bin_edges : list of float
+            List of bin edges, with each element having a min and max value.
+            Each element must contain two values
+        dac_value : list od float
+            List of dac values for each exposure bin
+        Returns
+        -------
+        exposure_times : `list`
+            A list of random exposure times, rounded to 1 decimal place.
+            The length of the list will be: (len(bin_edges) - 1) *
+            samples_per_bin.
+        dac_values : `list`
+        Raises
+        ------
+        ValueError
+            If fewer than two bin edges are provided.
+        """
+        exposure_times_list: list[NDArray[np.float_]] = []
+        bin_edges = np.array(bin_edges, dtype=float)
+
```

Comment:
You could consider adding a check in there that dac_values is len(bin_edges) - 1. You can do that with an assert statement that raises an info message for the user when they are not.

Reply:
It should actually be the length of bin_edges, because bin_edges is now a list of 2 items: [start, end]. But I will add and assert.